cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/mazha/CouchClient.java

```diff
@@ -75,6 +105,145 @@ public CouchClient(URI rootUri,
+        // Some assistance for TLSv1.2 support. Two things we check before we try to force TLSv1.2
+        // so that we don't interfere with any other custom configuration that may have been
+        // provided:
+        //   i) are we running on Android, and an old version of Android (api level < 20 does not
+        //      have TLSv1.2 enabled by default)
+        //   ii) does the default SSLContext have TLSv1.2 enabled or available
+        if (Misc.isRunningOnAndroid()) {
+
+            System.out.println("RUNNING ON ANDROID");
+            // This block catches all exceptions from TLSv1.2 checks, if we get exceptions we bail
+            // with a RuntimeException
+            try {
+                String tlsv1_2 = "TLSv1.2";
+                // Get the API level reflectively so we don't need to import classes only available
+                // in Android
+                int androidApiLevel = Class.forName("android.os.Build$VERSION").getField
+                        ("SDK_INT").getInt(null);
+                // If we are on an old version of Android and TLSv1.2 has not been enabled already
+                // on the default context then we add a special interceptor
+                if (androidApiLevel < 20) {
+                    // Check i has passed we are on an old version of Android
+
+                    // Get the default SSLContext
+                    SSLContext defSslCtx = SSLContext.getDefault();
+
+                    // Check the default protocols for TLSv1.2
+                    if (!Arrays.asList(defSslCtx.getDefaultSSLParameters().getProtocols())
+                            .contains(tlsv1_2) &&
+                            Arrays.asList(defSslCtx.getSupportedSSLParameters().getProtocols())
+                                    .contains(tlsv1_2)) {
+                        // Check ii has passed TLSv1.2 is not already enabled on the default context
+                        // but is supported on the default context so we can use it
+
+                        // In an ideal world we would also check the default SSLSocketFactory that
+                        // is set on the HttpsUrlConnection to see if that has been customized from
+                        // the default, but there isn't really a way to check this, equals is not
+                        // overridden on the SSLSocketFactory and there is no route back to the
+                        // SSLContext that generated the SSLSocketFactory.
+                        SSLContext sc = SSLContext.getInstance(tlsv1_2);
+                        sc.init(null, null, null);
+                        // We add this interceptor, but it could be overridden by a later user
+                        // provided interceptor, so this shouldn't change anyone's existing
+                        // customizations.
+                        SSLSocketFactory s = sc.getSocketFactory();
+                        return new Special(s);
```

Comment:
Can you remove the `Special` class (see below), and just use `SSLCustomizerInterceptor` directly?

Reply:
Yes, as above, done in 5b23adcf577db53b8ed66e4ab6b8ade7c8e4ff4a